File: courses/C# 2008 - Impacta/C# 2008 - M2/Instrutor/CSModII20090918/VB.NET2008 Mod2 Hanna/VBNet II/Módulo II/Laboratórios/Capitulo04_Laboratorio02/Oficina.BLL/VeiculoBLL.vb
Imports Oficina.Modelos
Imports Oficina.DAL
Imports System.Data.Common

Public Class VeiculoBLL
    Public Sub Incluir(ByVal veiculo As VeiculoInfo)
        If String.IsNullOrEmpty(veiculo.Placa) Then
            Throw New Exception("Informe a placa do veículo")

        ElseIf (String.IsNullOrEmpty(veiculo.Modelo)) Then
            Throw New Exception("Informe o modelo do veículo")

        ElseIf (String.IsNullOrEmpty(veiculo.Cor)) Then
            Throw New Exception("Informe a cor do veículo")

        ElseIf (String.IsNullOrEmpty(veiculo.Ano.ToString())) Then
            Throw New Exception("Informe o ano do veículo")

        End If
        Dim obj As New VeiculoDAL()
        obj.Incluir(veiculo)
    End Sub

    'Retorna um DataReader genérico que não depende do 
    'fabricante do servidor de dados
    Public Function PesquisarPorPlaca(ByVal placa As String) As DbDataReader

        If String.IsNullOrEmpty(placa) Then
            Throw New Exception("Informe uma placa")
        End If

        Dim obj As New VeiculoDAL()
        Return obj.PesquisarPorPlaca(placa)
    End Function

    Public Sub Alterar(ByVal veiculo As VeiculoInfo)
        If String.IsNullOrEmpty(veiculo.Placa) Then
            Throw New Exception("Informe a placa do veículo")

        ElseIf String.IsNullOrEmpty(veiculo.Modelo) Then
            Throw New Exception("Informe o modelo do veículo")

        ElseIf String.IsNullOrEmpty(veiculo.Cor) Then
            Throw New Exception("Informe a cor do veículo")

        ElseIf String.IsNullOrEmpty(veiculo.Ano.ToString()) Then
            Throw New Exception("Informe o ano do veículo")

        End If
        Dim obj As New VeiculoDAL()
        obj.Alterar(veiculo)
    End Sub

    Public Sub Excluir(ByVal veiculo As VeiculoInfo)
        If String.IsNullOrEmpty(veiculo.Placa) Then
            Throw New Exception("Informe a placa do veículo")

        End If
        Dim obj As New VeiculoDAL()
        obj.Excluir(veiculo)
    End Sub

End Class

File: courses/C# 2008 - Impacta/C# 2008 - M2/Instrutor/LaboratóriosAPAGAR/Capitulo06_Laboratorio02/Oficina.BLL/VeiculoBLL.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using System.Data.Common;
using Oficina.Modelos;
using Oficina.DAL;


namespace Oficina.BLL
{
    public class VeiculoBLL
    {
        public void Incluir(VeiculoInfo veiculo)
        { 
            if(string.IsNullOrEmpty(veiculo.Placa))
            {
                throw new Exception("Informe a placa do veículo");
            }
            else if(string.IsNullOrEmpty(veiculo.Modelo))
            {
                throw new Exception("Informe o modelo do veículo");
            }
            else if(string.IsNullOrEmpty(veiculo.Cor))
            {
                throw new Exception("Informe a cor do veículo");
            }
            else if(string.IsNullOrEmpty(veiculo.Ano.ToString()))
            {
                throw new Exception("Informe o ano do veículo");
            }
            VeiculoDAL obj = new VeiculoDAL();
            obj.Incluir(veiculo);
        }

        //Retorna um DataReader genérico que não depende do fabricante do servidor de dados
        public DbDataReader PesquisarPorPlaca(string placa)
        {
            if (string.IsNullOrEmpty(placa))
            {
                throw new Exception("Informe uma placa");
            }
            VeiculoDAL obj = new VeiculoDAL();
            return obj.PesquisarPorPlaca(placa);
        }

        public void Alterar(VeiculoInfo veiculo)
        {
            if (string.IsNullOrEmpty(veiculo.Placa))
            {
                throw new Exception("Informe a placa do veículo");
            }
            else if (string.IsNullOrEmpty(veiculo.Modelo))
            {
                throw new Exception("Informe o modelo do veículo");
            }
            else if (string.IsNullOrEmpty(veiculo.Cor))
            {
                throw new Exception("Informe a cor do veículo");
            }
            else if (string.IsNullOrEmpty(veiculo.Ano.ToString()))
            {
                throw new Exception("Informe o ano do veículo");
            }
            VeiculoDAL obj = new VeiculoDAL();
            obj.Alterar(veiculo);
        }

        public void Excluir(VeiculoInfo veiculo)
        {
            if (string.IsNullOrEmpty(veiculo.Placa))
            {
                throw new Exception("Informe a placa do veículo");
            }
            VeiculoDAL obj = new VeiculoDAL();
            obj.Excluir(veiculo);
        }

    }
}
